diff --git a/tensorflow/contrib/cmake/CMakeLists.txt b/tensorflow/contrib/cmake/CMakeLists.txt
index 23b31ae1dc..aba1173de8 100644
--- a/tensorflow/contrib/cmake/CMakeLists.txt
+++ b/tensorflow/contrib/cmake/CMakeLists.txt
@@ -405,6 +405,14 @@ if (tensorflow_ENABLE_GPU)
     ${CUDNN_INCLUDE}/cudnn.h
     DESTINATION ${tensorflow_source_dir}/third_party/gpus/cuda/include
   )
+  FILE(COPY
+    ${CUDA_TOOLKIT_TARGET_DIR}/extras/CUPTI/include/cupti.h
+    DESTINATION ${tensorflow_source_dir}/third_party/gpus/cuda/extras/CUPTI/include
+  )
+  FILE(COPY
+    ${tensorflow_PATH_NCCL_STATIC_LIB}/../include/nccl.h
+    DESTINATION ${tensorflow_source_dir}/third_party/nccl
+  )
 
   include_directories(${tensorflow_source_dir}/third_party/gpus)
   # add cuda libraries to tensorflow_EXTERNAL_LIBRARIES
